server/src/main/java/org/candlepin/dto/api/v1/UserDTO.java

```diff
@@ -0,0 +1,249 @@
+        super.populate(source);
+
+        this.setId(source.getId());
+        this.setUsername(source.getUsername());
+        this.setPassword(source.getPassword());
+        this.setSuperAdmin(source.isSuperAdmin());
+
+        return this;
+    }
+
+    /**
+     * Fetches this user's ID. If the ID has not been set, this method returns null.
+     *
+     * @return
+     *  This user's ID, or null if the ID has not been set
+     */
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets or clears the ID for this user.
+     *
+     * @param id
+     *  The ID to set for this user, or null to clear any previously set ID
+     *
+     * @return
+     *  a reference to this DTO
+     */
+    public UserDTO setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Fetches this user's username. If the username has not been set, this method returns null.
+     *
+     * @return
+     *  This user's username, or null if the username has not been set
+     */
+    public String getUsername() {
+        return this.username;
+    }
+
+    /**
+     * Sets or clears the username for this user.
+     *
+     * @param username
+     *  The username to set for this user, or null to clear any previously set username
+     *
+     * @return
+     *  a reference to this DTO
+     */
+    public UserDTO setUsername(String username) {
+        this.username = username;
+        return this;
+    }
+
+    /**
+     * Fetches the password set for this user. If the password hasn't been set, this method returns
+     * null.
+     * <p></p>
+     * <strong>
+     *  Note that for obvious reasons, this be serialized for output under any sane circumstances.
+     * </strong>
+     *
+     * @return
+     *  The password set for this user, or null if the password hasn't been set.
+     */
+    @JsonIgnore
+    public String getPassword() {
+        return this.password;
+    }
+
+    /**
+     * Sets or clears the plain-text password to use for this user.
+     *
+     * @param password
+     *  The password, in plain text, to set for this user, or null to clear any previously set
+     *  password
+     *
+     * @return
+     *  a reference to this DTO
+     */
+    public UserDTO setPassword(String password) {
+        // Impl note:
+        // While this would be a horrid practice in any other language, in Java, there's not a whole
+        // lot we can do about this. As soon as the plain-text password is in the JVM, we have no
+        // way of guaranteeing a (shadow) copy of that variable won't be stored in the garbage
+        // collector or whatever goofy optimization mapping Java is doing under the hood.
+        // As such, what is typically a very unsafe practice in most languages is par for the course
+        // here in Java land. This is really not a great deal worse than instantly hashing it as we
+        // do in the User entity given how it was used.
```

Comment:
Is it not possible to use a `char[]` as the password type, since an array can be overwritten when the password is no longer needed, and it won't stay in the string pool? (obviously not something for this PR, just a general thought)

Reply:
No, because the risk here isn't that we can't clear the string when we're done with it (though it helps minimize this potential for a memory-based attack), it's that if the GC runs while the char array is being maintained with a valid password, depending on the GC implementation, you could end up with another copy of the array in memory as the GC shuffles things around.

Either way, we're kinda stuck with a string here due to the interface we're using.

Reply:
I see